test/dynamo/test_torchxla_integration.py

```diff
@@ -125,9 +140,61 @@ def test_wrapper(self):
 
     return test_wrapper
 
+def training_compiler(gm, example_inputs):
+    @make_boxed_compiler
+    @disable
+    def fw_compiler(graph, inputs, *args, **kwargs):
+        return integration.extract_compiled_graph(graph, inputs)
+
+    @make_boxed_compiler
+    @disable
+    def bw_compiler(graph, inputs, *args, **kwargs):
+        return integration.extract_compiled_graph(graph, inputs)
+
+    return aot_module_simplified(gm, example_inputs, fw_compiler=fw_compiler, bw_compiler=bw_compiler)
+
+
+def model_iter_fn_train(mod, inputs):
+    outputs = mod(*inputs)
+    loss = outputs.mean()
+    loss.backward()
+
+    param_list = list(mod.parameters())
+    return [param.grad for param in param_list]
+
+def make_training_test(model_cls):
+    @maybe_skip_torchxla_test
+    def test_wrapper(self):
+        import torch_xla.core.xla_model as xm
+        import torch_xla
+        xla_dev = xm.xla_device()
+        model = model_cls()
+        inputs = model.get_random_inputs()
+
+        model = model.to(device=xla_dev)
+        inputs = tuple(inp.to(device=xla_dev) for inp in inputs)
+
+        # do baseline
+        baseline_model = copy.deepcopy(model)
+        baseline_inputs = copy.deepcopy(inputs)
+        expected_output = model_iter_fn_train(baseline_model, baseline_inputs)
+
+        compiler = training_compiler
+        optimize_ctx = torch._dynamo.optimize(compiler, nopython=False)
+        optimized_model_iter_fn = optimize_ctx(model_iter_fn_train)
+
+        actual_output = optimized_model_iter_fn(model, inputs)
+        print(f"expected_output:\n{expected_output}\nactual_output:\n{actual_output}")
+        assert allclose(expected_output, actual_output)
+
+    return test_wrapper
 
 class TorchXLAReuseGraphTest(unittest.TestCase):
     test_basic = make_reuse_graph_test(BasicModule)
     test_matmul = make_reuse_graph_test(MatmulModule)
     test_linear = make_reuse_graph_test(LinearModule)
     test_inplace_update = make_reuse_graph_test(ModuleInplaceUpdate)
+
+    test_training_linear = make_training_test(LinearModule)
+
+    test_training_maxpool = unittest.skip("max_pool2d does not work: https://github.com/pytorch/torchdynamo/issues/1837")(make_training_test(MaxPoolModule))
```

Comment:
is this fixed now, so you can unskip the test?

Reply:
yep, will do